packages/react/src/components/Link/index.tsx

```diff
@@ -2,35 +2,56 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 
+// @ts-ignore
+import { BrowserRouter, Link as RouterLink } from 'react-router-dom';
+
 export interface LinkProps
   extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
   linkRef?: React.Ref<HTMLAnchorElement>;
+  reactRouter?: false;
   variant?: 'button';
+  useRouter?: false;
 }
 
 const Link = ({
   children,
   linkRef,
   className,
   variant,
+  useRouter,
   ...other
-}: LinkProps) => (
-  <a
-    ref={linkRef}
-    className={classNames(className, {
-      Link: !variant,
-      'Button--primary': variant === 'button'
-    })}
-    {...other}
-  >
-    {children}
-  </a>
-);
+}: LinkProps) =>
+  useRouter ? (
+    <BrowserRouter>
```

Comment:
We don't want to include `BrowserRouter` for every single link...The idea of this feature would be to be able to:

```js
import { Link as CauldronLink } from '@deque/cauldron-react'
import { Link } from 'react-router-dom' 
 
const Foo = () => {
  return <CauldronLink component={Link} href="/about" />
}
```

Reply:
Yeah, we can't include multiple routers here. This would break our apps.

We also don't want to _only_ support React Router. We might decide to build an app with [`Wouter`](https://github.com/molefrog/wouter) and need its [`<Link/>`](https://github.com/molefrog/wouter/#link-hrefpath), or maybe [Next.js's `<Link/>`](https://nextjs.org/docs/api-reference/next/link) instead.

Harris' approach of passing a `component={SomeLinkComponent}` prop seems like the best solution to me.